clean_project/construct_dict_test.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+import json
+import tempfile
+import unittest
+
+from construct_dict import construct_dictionary
+
+
+class ConstructDictTest(unittest.TestCase):
+
+    def test_empty_yaml_file(self):
+        with tempfile.NamedTemporaryFile(suffix="JSON") as json_file:
+            self.assertEqual("Empty yaml file", construct_dictionary('test_data/config_empty_file.yaml', json_file.name))
+
+    def test_full_dictionary(self):
+        with tempfile.NamedTemporaryFile(suffix="JSON") as json_file:
```

Comment:
i'm not questioning the fact that you use a temporary file. i'm saying that you should modify the file while it's open. create the temp file with the delete flag set to false and then delete it manually at the end